File: 000androidpoints/src/main/java/com/xiaoming/androidpoints/activity/onactivityresult/BActivity.java
package com.xiaoming.androidpoints.activity.onactivityresult;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.xiaoming.androidpoints.R;

public class BActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_b);

        Button btn = (Button) findViewById(R.id.bt_b_back_a);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // 将先要传回的数据放到Intent里
                // 可以用putExtra()的方法，也可以用setXXX()的方法
                Intent intent = new Intent();
                intent.putExtra("respond", "从B返回");
                // 设置返回码和返回携带的数据
                setResult(Activity.RESULT_OK, intent);
                // RESULT_OK就是一个默认值，=-1，它说OK就OK吧
                finish();
            }
        });
    }
}

File: 010kotlin/src/main/java/com/xiaoming/a010kotlin/androidpoints/activitylifecycle/BActivity.kt
package com.xiaoming.a010kotlin.androidpoints.activitylifecycle

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.xiaoming.a010kotlin.R

//全透明
//A Activity被全透明B Activity覆盖后，只会执行onPause，不会执行onStop
class BActivity : AppCompatActivity() {
    companion object {
        const val TAG = "BActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "lifecycle#onCreate")
        setContentView(R.layout.activity_b)
    }

    override fun onRestart() {
        super.onRestart()
        Log.d(TAG, "lifecycle#onRestart")
    }

    override fun onStart() {
        super.onStart()
        Log.d(TAG, "lifecycle#onStart")
    }

    override fun onResume() {
        super.onResume()
        Log.d(TAG, "lifecycle#onResume")
    }

    override fun onPause() {
        super.onPause()
        Log.d(TAG, "lifecycle#onPause")
    }

    override fun onStop() {
        super.onStop()
        Log.d(TAG, "lifecycle#onStop")
    }
    override fun onDestroy() {
        super.onDestroy()
        Log.d(TAG, "lifecycle#onDestroy")
    }
}
